### PR TITLE
Load countries directory select by store

### PR DESCRIPTION
**Pull request by fplantinet**
Fix countries allowed in checkout 

Source: https://blog.fabian-blechschmidt.de/all-countries-in-checkout-although-allowed_countries-set-correctly/

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php [hunk 1]
```diff
@@ -45,10 +45,15 @@ public function getAllOptions()
...
+            if (Mage::app()->getStore()->isAdmin()) {
+                $collection = Mage::getModel('directory/country')->getResourceCollection();
+            } else {
+                $collection = Mage::getModel('directory/country')->getResourceCollection()->loadByStore();
+            }
             $options = $collection->toOptionArray();
+
             if (Mage::app()->useCache('config')) {
-                Mage::app()->saveCache(serialize($options), $cacheKey, array('config'));
+                Mage::app()->saveCache(serialize($options), $cacheKey, ['config']);
```
Comment (edannenberg) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php]: If this is a purely cosmetic change I would recommend removing it. Same for the introduced new line. Less changes = better.

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php [hunk 2]
```diff
@@ -45,10 +45,15 @@ public function getAllOptions()
         if (Mage::app()->useCache('config') && $cache = Mage::app()->loadCache($cacheKey)) {
             $options = unserialize($cache);
         } else {
-            $collection = Mage::getModel('directory/country')->getResourceCollection();
+            if (Mage::app()->getStore()->isAdmin()) {
```
Comment (edannenberg) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php]: Refactor suggestion:

            $collection = Mage::getModel('directory/country')->getResourceCollection();
            if (!Mage::app()->getStore()->isAdmin())
                  $collection->loadByStore();

This way the patch becomes additive only.

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php (hunk 1 above)
Comment (fplantinet) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php]: ok thanx!

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php (hunk 2 above)
Comment (fplantinet) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Source/Countryofmanufacture.php]: ok, i update commit ! thanx

**Pull request merged**